src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/AbstractReferenceFinder.cs

```diff
@@ -720,13 +721,23 @@ public override Task<ImmutableArray<Project>> DetermineProjectsToSearchAsync(ISy
                 : SpecializedTasks.EmptyImmutableArray<Project>();
         }
 
-        public override Task<ImmutableArray<Document>> DetermineDocumentsToSearchAsync(
+        public async override Task<ImmutableArray<Document>> DetermineDocumentsToSearchAsync(
             ISymbol symbol, Project project, IImmutableSet<Document> documents,
             FindReferencesSearchOptions options, CancellationToken cancellationToken)
         {
-            return symbol is TSymbol && CanFind((TSymbol)symbol)
-                ? DetermineDocumentsToSearchAsync((TSymbol)symbol, project, documents, options, cancellationToken)
-                : SpecializedTasks.EmptyImmutableArray<Document>();
+            if (symbol is TSymbol targetSymbol && CanFind(targetSymbol))
+            {
+                var documentsToSearch = await DetermineDocumentsToSearchAsync(targetSymbol, project, documents, options, cancellationToken).ConfigureAwait(false);
```

Comment:
i'm worried about this.  because this may affect cascading.  i.e. there's a document we need to search because it opens up information about new symbols that find new cascades *in other documents*.

Reply:
Reverted. See comment above